services/QuillLMS/app/models/classrooms_teacher.rb

```diff
@@ -14,15 +15,18 @@
 #
 # Indexes
 #
-#  index_classrooms_teachers_on_classroom_id             (classroom_id)
-#  index_classrooms_teachers_on_role                     (role)
-#  index_classrooms_teachers_on_user_id                  (user_id)
-#  unique_classroom_and_user_ids_on_classrooms_teachers  (user_id,classroom_id) UNIQUE
+#  index_classrooms_teachers_on_classroom_id            (classroom_id)
+#  index_classrooms_teachers_on_role                    (role)
+#  index_classrooms_teachers_on_user_id                 (user_id)
+#  unique_user_class_deleted_at_on_classrooms_teachers  (user_id,classroom_id,deleted_at) UNIQUE
 #
 class ClassroomsTeacher < ApplicationRecord
   belongs_to :user
   belongs_to :classroom, touch: true
 
+  validates :user_id, uniqueness: { scope: [:classroom_id, :deleted_at], message: "user_id, classroom_id, and deleted_at must be collectively unique"}
```

Comment:
The custom message is important here; the default message is misleading: `Validation failed: User has already been taken>`